hudi-common/src/main/java/org/apache/hudi/common/config/ConfigProperty.java

```diff
@@ -95,33 +99,46 @@ Option<Function<HoodieConfig, Option<T>>> getInferFunc() {
     return inferFunction;
   }
 
+  public void checkValues(String value) {
+    if (validValues != null && !validValues.isEmpty() && !validValues.contains(value)) {
+      throw new IllegalArgumentException(
+          "The value of " + key + " should be one of "
+              + String.join(",", validValues) + ", but was " + value);
+    }
+  }
+
   public List<String> getAlternatives() {
     return Arrays.asList(alternatives);
   }
 
   public ConfigProperty<T> withDocumentation(String doc) {
     Objects.requireNonNull(doc);
-    return new ConfigProperty<>(key, defaultValue, doc, sinceVersion, deprecatedVersion, inferFunction, alternatives);
+    return new ConfigProperty<>(key, defaultValue, doc, sinceVersion, deprecatedVersion, inferFunction, validValues, alternatives);
+  }
+
+  public ConfigProperty<T> withValidValues(List<String> validValues) {
```

Comment:
Why List? Why not Set?

Reply:
fixed

Reply:
Guess this is not for performance concern, and more for easier API usage. varargs actually works better. see the other API `withAlternatives(String... alternatives)`

Reply:
From that perspective, I agree list is more consistent. Perf does not matter much for just a few elements. 
@wzx140 sorry for the trouble, but can you revert it back to `withValidValues(String... validValues)`

Reply:
@codope Fixed. Use `withValidValues(String... validValues)` and store them with set.